File: wiki-backend/src/main/java/com/internal/team/wiki/domain/user/hashing/Hashing.java
package com.internal.team.wiki.domain.user.hashing;

import java.security.MessageDigest;
import java.security.NoSuchAlgorithmException;

public class Hashing implements HashingI {

    protected Hashing() {
    }

    @Override
    public String generateSHA256Hash(String text) {
        try {
            MessageDigest messageDigest = MessageDigest.getInstance("SHA-256");
            messageDigest.update(text.getBytes());
            return bytesToHex(messageDigest.digest());
        } catch (NoSuchAlgorithmException e) {
            throw new RuntimeException(e);
        }
    }

    private String bytesToHex(byte[] bytes) {
        StringBuilder builder = new StringBuilder();
        for (byte b : bytes) {
            builder.append(String.format("%02x", b));
        }
        return builder.toString();
    }
}

File: wiki-backend/src/main/java/com/internal/team/wiki/global/hashing/Hashing.kt
package com.internal.team.wiki.global.hashing

// 인터페이스는 객체의 설계를 정의하는 역할을 할 뿐, 실제 구현을 가지지 않기 때문에 인스턴스를 생성할 수 없음
interface Hashing {
    fun generateSHA256Hash(text: String): String
}